File: radio/src/main/java/com/dreampany/radio/data/source/room/StationRoomDataSource.java
package com.dreampany.radio.data.source.room;

import com.dreampany.radio.data.misc.StationMapper;
import com.dreampany.radio.data.model.Station;
import com.dreampany.radio.data.source.api.StationDataSource;
import io.reactivex.Maybe;

import javax.inject.Singleton;
import java.util.List;

/**
 * Created by Hawladar Roman on 1/9/2019.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */

@Singleton
public class StationRoomDataSource implements StationDataSource {

    private final StationMapper mapper;
    private final StationDao dao;

    public StationRoomDataSource(StationMapper mapper,
                                 StationDao dao) {
        this.mapper = mapper;
        this.dao = dao;
    }

    @Override
    public boolean isEmpty() {
        return false;
    }

    @Override
    public Maybe<Boolean> isEmptyRx() {
        return null;
    }

    @Override
    public int getCount() {
        return 0;
    }

    @Override
    public Maybe<Integer> getCountRx() {
        return null;
    }

    @Override
    public boolean isExists(Station station) {
        return false;
    }

    @Override
    public Maybe<Boolean> isExistsRx(Station station) {
        return null;
    }

    @Override
    public long putItem(Station station) {
        return 0;
    }

    @Override
    public Maybe<Long> putItemRx(Station station) {
        return null;
    }

    @Override
    public List<Long> putItems(List<Station> stations) {
        return null;
    }

    @Override
    public Maybe<List<Long>> putItemsRx(List<Station> stations) {
        return null;
    }

    @Override
    public Station getItem(long id) {
        return null;
    }

    @Override
    public Maybe<Station> getItemRx(long id) {
        return null;
    }

    @Override
    public List<Station> getItems() {
        return null;
    }

    @Override
    public Maybe<List<Station>> getItemsRx() {
        return null;
    }

    @Override
    public List<Station> getItems(int limit) {
        return null;
    }

    @Override
    public Maybe<List<Station>> getItemsRx(int limit) {
        return null;
    }
}

File: radio/src/main/java/com/dreampany/radio/data/source/room/StationRoomDataSource.kt
package com.dreampany.radio.data.source.room

import com.dreampany.radio.data.model.Station
import com.dreampany.radio.data.source.api.StationDataSource
import com.dreampany.radio.data.source.room.dao.StationDao

/**
 * Created by roman on 2/11/20
 * Copyright (c) 2020 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
class StationRoomDataSource(
     private val dao: StationDao
) : StationDataSource {
    override suspend fun write(input: Station): Long {
        TODO("Not yet implemented")
    }

    override suspend fun write(inputs: List<Station>): List<Long>? {
        TODO("Not yet implemented")
    }

    override suspend fun reads(): List<Station>? {
        TODO("Not yet implemented")
    }

    override suspend fun readsByCountry(
        country: String,
        order: Station.Order,
        offset: Long,
        limit: Long
    ): List<Station>? {
        TODO("Not yet implemented")
    }

    override suspend fun readsByCountryCode(
        countryCode: String,
        order: Station.Order,
        offset: Long,
        limit: Long
    ): List<Station>? {
        TODO("Not yet implemented")
    }

    override suspend fun readsByLanguage(
        language: String,
        order: Station.Order,
        offset: Long,
        limit: Long
    ): List<Station>? {
        TODO("Not yet implemented")
    }

    override suspend fun readsByTopClick(
        order: Station.Order,
        offset: Long,
        limit: Long
    ): List<Station>? {
        TODO("Not yet implemented")
    }

    override suspend fun readsByTopVote(
        order: Station.Order,
        offset: Long,
        limit: Long
    ): List<Station>? {
        TODO("Not yet implemented")
    }

    override suspend fun readsByLastClick(
        order: Station.Order,
        offset: Long,
        limit: Long
    ): List<Station>? {
        TODO("Not yet implemented")
    }

    override suspend fun readsByLastChange(
        order: Station.Order,
        offset: Long,
        limit: Long
    ): List<Station>? {
        TODO("Not yet implemented")
    }

    override suspend fun searchByName(
        name: String,
        order: Station.Order,
        offset: Long,
        limit: Long
    ): List<Station>? {
        TODO("Not yet implemented")
    }

    override suspend fun searchByTag(
        tag: String,
        order: Station.Order,
        offset: Long,
        limit: Long
    ): List<Station>? {
        TODO("Not yet implemented")
    }
}